sklearn/utils/validation.py

```diff
@@ -103,6 +103,9 @@ def _is_arraylike(x):
 
 def _num_samples(x):
     """Return number of samples in array-like x."""
+    if hasattr(x, 'fit'):
+        # Don't get num_samples from an ensembles length!
+        raise TypeError('Expected dataset, but found estimator {!r}'.format(x))
```

Comment:
Ah of course, thanks! I forget that every time (thankfully travis tests in py2.6 nowadays; it used to only alert us after merge).